src/Hpack/Config.hs

```diff
@@ -1343,7 +1343,7 @@ inferModules ::
   -> IO (Section b)
 inferModules dir packageName_ getMentionedModules getInferredModules fromData fromConditionals = fmap removeConditionalsThatAreAlwaysFalse . traverseSectionAndConditionals
   (fromConfigSection fromData [pathsModuleFromPackageName packageName_])
-  (fromConfigSection (\ [] -> fromConditionals) [])
+  (fromConfigSection (\ _ -> fromConditionals) [])
```

Comment:
@bergmark do you know for what reason this change was necessary?  Did you get a test failure?

Reply:
The code is somewhat involved, but I think it's statically guaranteed that this list is always empty.  Looking at the relevant parts of `fromConfigSection`:

```haskell
  where
    fromConfigSection fromConfig pathsModule_ ... = do
      ...
      let
        ...
        pathsModule = (pathsModule_ \\ outerModules) \\ mentionedModules
        r = fromConfig pathsModule inferableModules conf
      ...
```
Substituting `fromConfig` with `(\ [] -> fromConditionals)` and `pathsModule_` with `[]`:
```haskell
      ...
      let
        ...
        pathsModule = ([] \\ outerModules) \\ mentionedModules
        r = (\ [] -> fromConditionals) pathsModule inferableModules conf
      ...
```

=>
```haskell
      ...
      let
        ...
        pathsModule = []
        r = (\ [] -> fromConditionals) pathsModule inferableModules conf
      ...
```
=>
```haskell
      ...
      let
        ...
        r = (\ [] -> fromConditionals) [] inferableModules conf
      ...
```
we see that the lambda is only ever called with an empty list.

Reply:
Umm, this is actually because of `-Wincomplete-uni-patterns` being added to `-Wall`.   I'm still going to disable that warning instead, as matching on `[]` communicates that static guarantee to the reader.